### PR TITLE
Work on examples

### PR DESCRIPTION
**Pull request by jschumme**
# Justification

- When starting the DataPlugin creation with a sample file, we copy the `hello_world` template and the chosen sample file to the project. This PR should rename the hello_world.py to the correct DataPlugin name and replace the referenced Example.csv in the script by the chosen sample file:

  image

- Improved the existing csv examples

# Implementation

Added the following functionality to the DataPlugin class:
- replaceStringInScript(substr: string, newSubStr: string): void
- renameDataPluginScript(newName: string): void

# Testing

Added two unit tests

### CODE REVIEW

File: examples/csv_read_with_callback_loading/csv_read_with_callback_loading.py [hunk 1]
```diff
@@ -14,8 +14,11 @@ class Plugin:
     def read_store(self, parameter):
         file_path = os.path.realpath(parameter["file"])
 
-        with open(file_path, newline="") as csvfile:
+        with open(file_path, newline="", encoding="utf8") as csvfile:
```
Comment (LukasKall) [inline examples/csv_read_with_callback_loading/csv_read_with_callback_loading.py:17]: I think it should be `encoding="utf-8"`.

File: examples/csv_read_with_callback_loading/csv_read_with_callback_loading.py [hunk 2]
```diff
@@ -14,8 +14,11 @@ class Plugin:
     def read_store(self, parameter):
         file_path = os.path.realpath(parameter["file"])
 
-        with open(file_path, newline="") as csvfile:
+        with open(file_path, newline="", encoding="utf8") as csvfile:
             tab_delimiter = "\t"
+            # skip the first 4 lines
+            # for i in range(4):
```
Comment (LukasKall) [inline examples/csv_read_with_callback_loading/csv_read_with_callback_loading.py:20]: Is it intended to be commented out?

File: examples/csv_read_with_direct_loading/csv_read_with_direct_loading.py [hunk 3]
```diff
@@ -15,8 +15,11 @@ class Plugin:
     def read_store(self, parameter):
         file_path = os.path.realpath(parameter["file"])
 
-        with open(file_path, newline="") as csvfile:
+        with open(file_path, newline="", encoding="utf8") as csvfile:
```
Comment (LukasKall) [inline examples/csv_read_with_direct_loading/csv_read_with_direct_loading.py:18]: I think it should be `encoding="utf-8"`.

File: examples/csv_read_with_direct_loading/csv_read_with_direct_loading.py [hunk 4]
```diff
@@ -15,8 +15,11 @@ class Plugin:
     def read_store(self, parameter):
         file_path = os.path.realpath(parameter["file"])
 
-        with open(file_path, newline="") as csvfile:
+        with open(file_path, newline="", encoding="utf8") as csvfile:
             tab_delimiter = "\t"
+            # skip the first 4 lines
+            # for i in range(4):
```
Comment (LukasKall) [inline examples/csv_read_with_direct_loading/csv_read_with_direct_loading.py:21]: Is it intended to be commented out?

File: src/dataplugin.ts [hunk 5]
```diff
@@ -65,6 +65,27 @@ class DataPlugin {
...
+        this.scriptPath = newScriptPath;
+    }
+
+    /**
+     * @param substr A String that is to be replaced by newSubstr
+     * @param newSubStr Replacement string
+     */
+    public replaceStringInScript(substr: string, newSubStr: string): void {
+        const scriptPath = this.scriptPath;
+        let content: string = fs.readFileSync(scriptPath, { encoding: 'utf8' });
```
Comment (LukasKall) [inline src/dataplugin.ts:84]: Can the type annotation be emitted? Or does the `readFileSync` don't provide any type information?

Comment (jschumme) [reply src/dataplugin.ts:84]: yes right, the type annotation is redundant 👍

File: examples/csv_read_with_direct_loading/csv_read_with_direct_loading.py (hunk 4 above)
Comment (jschumme) [reply examples/csv_read_with_direct_loading/csv_read_with_direct_loading.py:21]: yes it should only complement the example and might be useful.

File: examples/csv_read_with_callback_loading/csv_read_with_callback_loading.py (hunk 2 above)
Comment (jschumme) [reply examples/csv_read_with_callback_loading/csv_read_with_callback_loading.py:20]: yes it should only complement the example and might be useful.

**Review by LukasKall — APPROVED** [review]

**Review by EvaTi — APPROVED** [review]

**Pull request merged**